### ISSUE TITLE
SchemaError when trying to apply resources

### ISSUE THREAD

**Issue by discordianfish** [rancher/k3s#409]
**Describe the bug**
Getting this error, no matter what resource I'm trying to apply:

```error: SchemaError(io.k8s.api.core.v1.CinderVolumeSource): invalid object doesn't have additional properties```

Using `--validate=false` works around this.

**To Reproduce**
Not sure, I've just upgraded from 0.3 to 0.4.

**Comment by a-teisseire** [comment]
By upgrading K3s to 0.4, the underlying Kubernetes API got updated to v1.14.0.
The issue you're facing is because your kubectl is not up to date.
To check, you can type: `kubectl version`

In the best case, they have to be the same. If they're not, the Kubernetes spec explains that your kubectl can lag behind one minor version maximum.

**Comment by iameli** [comment]
Something I got bit with just in case it helps anyone else: Docker for Mac rather aggressively "upgrades" `/usr/local/bin/kubectl` to its preferred version when it boots up, so even if you've manually installed a later version recently it's worth checking that you didn't get downgraded.

**Comment by ibuildthecloud** [comment]
@discordianfish can you confirm that upgrading your kubectl to make k3s version worked?

**Comment by mjpitz** [comment]
@ibuildthecloud : I ran into the same issue due to the same reason. Verified that upgrading kubectl resolves this issue.

```
$ kubectl apply -f k8s/kubernetes-dashboard.yaml 
error: SchemaError(io.k8s.api.core.v1.CinderVolumeSource): invalid object doesn't have additional properties

$ brew upgrade kubectl
....

$ kubectl apply -f k8s/kubernetes-dashboard.yaml 
serviceaccount/kubernetes-dashboard created
clusterrolebinding.rbac.authorization.k8s.io/kubernetes-dashboard created
deployment.apps/kubernetes-dashboard created
service/kubernetes-dashboard created

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-19T22:12:47Z", GoVersion:"go1.12.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1-k3s.4", GitCommit:"52f3b42401c93c36467f1fd6d294a3aba26c7def", GitTreeState:"clean", BuildDate:"2019-04-15T22:13+00:00Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/arm64"}
```

**Comment by bcho** [comment]
I can reproduce with applying kubernetes dashboard's yaml with kubectl 1.10:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-26T16:55:54Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1-k3s.4", GitCommit:"52f3b42401c93c36467f1fd6d294a3aba26c7def", GitTreeState:"clean", BuildDate:"2019-04-15T22:13+00:00Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```

resolved with upgrading kubectl to 1.14 (the latest one):

```
$ ./kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1-k3s.4", GitCommit:"52f3b42401c93c36467f1fd6d294a3aba26c7def", GitTreeState:"clean", BuildDate:"2019-04-15T22:13+00:00Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by discordianfish** [comment]
Yes, sorry about the noise!
